apis/Google.Cloud.Metadata.V1/Google.Cloud.Metadata.V1.IntegrationTests/MetadataFixtureBase.cs

```diff
@@ -63,14 +63,30 @@ protected MetadataFixtureBase()
 
             Environment.SetEnvironmentVariable(EmulatorEnvironmentVariable, $"localhost:{port}");
 
-            emulatorProcess = Process.Start("python", $"{emulatorFilePath} --test --port {port}");
+            var startInfo = new ProcessStartInfo("python", $"{emulatorFilePath} --test --port {port}");
+            startInfo.RedirectStandardError = true;
+            startInfo.RedirectStandardOutput = true;
+            startInfo.UseShellExecute = false;
+            emulatorProcess = Process.Start(startInfo);
             emulatorProcess.EnableRaisingEvents = true;
             emulatorProcess.Exited += EmulatorProcess_Exited;
         }
 
         private void EmulatorProcess_Exited(object sender, EventArgs e)
         {
-            throw new InvalidOperationException("The metadata server emulator exited with code " + emulatorProcess.ExitCode);
+            var message = $"The metadata server emulator exited with code {emulatorProcess.ExitCode}";
+
+            var output = emulatorProcess.StandardOutput.ReadToEnd();
```

Comment:
Waiting until the end until we read anything can end up with it hanging due to a full buffer.
Maybe use the ErrorDataReceived and OutputDataReceived events instead?